src/commands/timelinecommands.cpp

```diff
@@ -1202,6 +1223,25 @@ void RemoveTransitionByTrimOutCommand::undo()
     if (m_clipIndex > 0) {
         LOG_DEBUG() << "trackIndex" << m_trackIndex << "clipIndex" << m_clipIndex << "delta" << m_delta;
         m_model.addTransitionByTrimIn(m_trackIndex, m_clipIndex, m_delta);
+        // Copy properties from old transition to new transition
+        std::unique_ptr<Mlt::ClipInfo> clipInfo = m_model.getClipInfo(m_trackIndex, m_clipIndex);
+        std::unique_ptr<Mlt::Service> oldService(new Mlt::Producer(MLT.profile(), "xml-string",
+                                                                   m_xml.toUtf8().constData()));
+        while (oldService && oldService->is_valid()) {
+            if (oldService->type() == mlt_service_transition_type) {
+                Mlt::Transition oldTransition(*oldService);
+                std::unique_ptr<Mlt::Service> newService(new Mlt::Producer(clipInfo->producer));
+                while (newService && newService->is_valid()) {
+                    if (newService->type() == mlt_service_transition_type &&
+                            QString(oldService->get("mlt_service")) == QString(newService->get("mlt_service"))) {
+                        newService->inherit(*oldService.get());
+                        break;
+                    }
+                    newService.reset(newService->producer());
```

Comment:
Given the above simplification this becomes `newService = newService.producer();`

Reply:
Changed to this to avoid memory leak:

```
Mlt::Service *tmpNewService = newService.producer();
newService = Mlt::Producer(*tmpNewService);
delete tmpNewService;
```